cmake/compile_tests/cplusplus20.cpp

```diff
@@ -14,12 +14,21 @@
 //
 //@HEADER
 
-#include <type_traits>
+#include <concepts>
 
-int main() {
-  // _v versions of type traits were added in C++17
-  if constexpr (std::is_same_v<double, int>)
-    return 0;
-  else
-    return 1;
+// consteval specifier
+consteval int sqr(int n) { return n * n; }
+constexpr int r = sqr(100);
```

Comment:
Even better if you check the result ?
```suggestion
static_assert(sqr(100) == 10000);
```